src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -201,13 +201,31 @@ def check_lazy(*args: Any, **kwargs: Any) -> Any:
                     returned_value = Index(returned_value, convert_to_lazy=False)
                 # Some methods also return a tuple with a pandas Index, so convert the tuple's first item to a modin Index
                 # Examples of this are `_get_indexer_strict` and `sort_values`
+
+                # We want to return a native pandas series for value counts because our loc/iloc functionality relies on it
+                # If we returned a snowpark pandas series, this would result in querying data many times
+                elif func_name == "value_counts":
```

Comment:
I'd like to learn more about this. Ideally, we should try to avoid this case.

Reply:
Yep sure. See [here](https://github.com/snowflakedb/snowpark-python/blob/68a2051b8fa5a5b8f436487cff2c3e6610d8eaaa/src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py#L778)

Reply:
I would prefer updating the code in indexing.py to call `to_pandas` there. Or you could do `return pd.Index(value_counts, convert_to_lazy=False)`. The point is to make sure the `value_counts` always return a pd.Index.

Reply:
`value_counts` is supposed to return a `Series`, since the method is unimplemented and the existing functionality relies on it beiing a native_series, I chose not to make any changes there in this PR. I was thinking that this would be something to do when we implement `value_counts`, and we have it return a snowpark pandas series

Reply:
oh, right, We should keep it consistent to return pd.Series. So how about add to_pandas in the indexing.py?